internal/hcsoci/layers.go

```diff
@@ -120,10 +120,18 @@ func MountContainerLayers(ctx context.Context, layerFolders []string, guestRoot
 		if uvm.OS() == "windows" {
 			options := &hcsschema.VirtualSmbShareOptions{
 				ReadOnly:            true,
-				PseudoOplocks:       true,
 				TakeBackupPrivilege: true,
 				CacheIo:             true,
 				ShareRead:           true,
+				PseudoDirnotify:     true,
+				NoLocks:             true,
+			}
+			if saveCapable {
+				// TODO(ambarve): If some other option disables
+				// NoDirectMap while saveastemplate is true then we
+				// should fail.
```

Comment:
Maybe we should add a condition check in our annotation verification then that a UVM cannot be fully physically backed when saveastemplate is specified?